EpsilonAuxBmsV2/Core/Src/Tasks/OrionInterfaceTask.c

```diff
@@ -0,0 +1,224 @@
+If a contactor is open, but it is allowed to be closed, the startupDone member of ContactorState indicates the startup sequence has finished, and Orion was the reason the contactor is open, Orion Interface will request that the contactor is closed.
+Will only ever be charge or discharge. If common is open after startup, the car would have tripped.
+When opening a contactor, it temporarily raises the gatekeeper task’s priority to RealTime (highest priority)
+*/
+void orionInterface(OrionCanInfo* message)
+{
+    if (auxBmsContactorState.orionHappyForStartup) {
+        osStatus_t status = osMessageQueueGet(orionInterfaceQueue, message, NULL, ORION_QUEUE_TIMEOUT);
+        //status = osMessageQueueGet(orionInterfaceQueue, message, NULL, ORION_QUEUE_TIMEOUT);
+
+
+        uint8_t shouldDisconnectContactors = 0;
+        // Read Orion charge and discharge enable GPIO
+        uint8_t orionDischargeEnableSense = HAL_GPIO_ReadPin(ORION_DISCHARGE_ENABLE_SENSE_GPIO_Port, ORION_DISCHARGE_ENABLE_SENSE_Pin);
+        uint8_t orionChargeEnableSense = HAL_GPIO_ReadPin(ORION_CHARGE_ENABLE_SENSE_GPIO_Port, ORION_CHARGE_ENABLE_SENSE_Pin);
+        // Set aux status and aux trip to default values
+        localAuxStatus = (AuxStatus)
+        {
+            0
+        };
+        localAuxStatus.allowCharge = 1;
+        localAuxStatus.allowDischarge = 1;
+        localAuxTrip = (AuxTrip)
+        {
+            0
+        };
+
+        if (status == osErrorTimeout) // If waiting on the queue times out
+        {
+            //Set orion can message recieved recently to 0
+            shouldDisconnectContactors = 1;
+            localAuxTrip.tripDueToOrionMessageTimeout = 1;
+            localAuxStatus.orionCanReceivedRecently = 0;
+        }
+        else
+        {
+            // Determine trip conditions and contactor settings based on Orion CAN
+            localAuxStatus.orionCanReceivedRecently = 1;
+            updateAllowChargeAndAllowDischarge(message, &localAuxStatus);
+            updateAuxTrip(message, &localAuxTrip);
+            localAuxStatus.dischargeShouldTrip = checkDischargeTrip(localAuxTrip);
+            localAuxStatus.chargeShouldTrip = checkChargeTrip(localAuxTrip);
+            shouldDisconnectContactors = localAuxTrip.protectionTrip
+                                        || localAuxStatus.dischargeShouldTrip
+                                        || localAuxStatus.chargeShouldTrip;
+        }
+
+        // Determine trip conditions and contactor settings based on Orion GPIO
+        shouldDisconnectContactors |= (!orionDischargeEnableSense && !orionChargeEnableSense);
+
+        // Set auxStatus if a trip is to occur (due to GPIO or CAN messages)
+        if (shouldDisconnectContactors)
+        {
+            localAuxStatus.allowCharge = 0;
+            localAuxStatus.allowDischarge = 0;
+
+            if(auxBmsContactorState.startupDone) {
+                localAuxStatus.strobeBmsLight = 1;
```

Comment:
We only turning on strobe lights if startup sequence finished? Why?

Reply:
because transient responses of the contactor closing makes it seem as though we should trip (especifically protection trip as current shuffles around), but we are ignoring those, so it does not make sense for us to turn on the strobe lights if we arent tripping

Reply:
I double checked myself and the strobebms turns on for about a quarter of a second on startup and it was always a protection trip when we close common, which is normal

Reply:
Sounds good.